packages/spidermonkey.rb

```diff
@@ -0,0 +1,55 @@
+require 'package'
+
+class Spidermonkey < Package
+  description 'SpiderMonkey is Mozilla\'s JavaScript engine written in C and C++.'
+  homepage 'https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey'
+  version '84.0'
+  compatibility 'all'
+  source_url 'https://archive.mozilla.org/pub/firefox/releases/84.0/source/firefox-84.0.source.tar.xz'
+  source_sha256 '23273ef0165b243f5d0908c38e7854d38070282c9b526e8d93b7503cd5f69138'
+
+  depends_on 'yasm'
+  depends_on 'rust' => :build
+  
+  def self.prebuild
+    puts
+    puts "Building spidermonkey requires at least 6.4 GB of space available on /mnt/stateful_partition.".lightblue
+    puts "It WILL fail without this space. Press Ctrl + C to cancel.".lightblue
+    puts
+    sleep(5)
+  end
+
+  def self.build
+    ENV['LC_ALL'] = 'C' # https://bugs.gentoo.org/746215
+    ENV['MACH_USE_SYSTEM_PYTHON'] = "#{CREW_PREFIX}/bin/python3"
+    ENV['CC'] = 'gcc' # Defaults to clang, but configure.py complains about oudated libstdc++ when using clang
+    ENV['CXX'] = 'g++'
+    ENV['AR'] = 'ar'
+    ENV['AS'] = 'as'
+    ENV['RANLIB'] = 'ranlib'
+    ENV['NM'] = 'nm'
+    ENV['LD'] = 'ld.gold'
+    Dir.chdir 'build' do
+      system "#{CREW_PREFIX}/bin/python3 \
+            ../configure.py --prefix=/usr/local \
+            --libdir=/usr/local/lib64 \
+            --host=x86_64-cros-linux-gnu \
+            --target=x86_64-cros-linux-gnu \
```

Comment:
We cannot hardcode this for one architecture:
```shell
      system "#{CREW_PREFIX}/bin/python3 \
            ../configure.py --prefix=#{CREW_DEST_PREFIX} \
            --libdir=#{CREW_DEST_LIB_PREFIX} \
            --host=#{ARCH}-cros-linux-gnu \
            --target=#{ARCH}-cros-linux-gnu \
```

Reply:
That actually isn't going to work either.

One way to deal with that:

```
  case ARCH
  when 'aarch64', 'armv7l'
  @gnuabi = 'gnueabihf'
  when 'x86_64', 'i686'
  @gnuabi = 'gnu'
  end

     def self.build
          system "python3.9 \
            ../configure.py --prefix=#{CREW_DEST_PREFIX} \
            --libdir=#{CREW_DEST_LIB_PREFIX} \
            --host=#{ARCH}-cros-linux-#{@gnuabi} \
            --target=#{ARCH}-cros-linux-#{@gnuabi} \
```

Reply:
Oh, and don't use CREW_DEST_PREFIX until in install...